From e2edf1fce0f84d6451ef4a70e27616c1d74887e8 Mon Sep 17 00:00:00 2001
From: Sahana Rao <sahanas@google.com>
Date: Fri, 30 Sep 2022 21:34:45 +0100
Subject: [PATCH] Set intent priority for PICK_IMAGES

Previously we were incorrectly setting the priority on the activity tag
instead of intent-filter tag. Changed the priority tag to be set
correctly on the intent-filter for PICK_IMAGES intent action

Bug: 246301995
Test: atest CtsPhotoPickerTest
Test: atest
android.photopicker.cts.ActionPickImagesOnlyTest#testPhotoPickerIntentDelegation

Change-Id: Iffcb615fee4df2e28874db9dd66642cccb81becc
(cherry picked from commit db654251f2754d1a4778111eace8086bc8b44959)
Merged-In: Iffcb615fee4df2e28874db9dd66642cccb81becc
---
 AndroidManifest.xml | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/AndroidManifest.xml b/AndroidManifest.xml
index 788df5d2..6cdda123 100644
--- a/AndroidManifest.xml
+++ b/AndroidManifest.xml
@@ -163,15 +163,14 @@
                 android:icon="@mipmap/picker_app_icon"
                 android:theme="@style/PickerDefaultTheme"
                 android:exported="true"
-                android:excludeFromRecents="true"
-                android:priority="100" >
-            <intent-filter>
+                android:excludeFromRecents="true" >
+            <intent-filter android:priority="100" >
                 <action android:name="android.provider.action.PICK_IMAGES" />
                 <category android:name="android.intent.category.DEFAULT" />
                 <data android:mimeType="image/*" />
                 <data android:mimeType="video/*" />
             </intent-filter>
-            <intent-filter>
+            <intent-filter android:priority="100" >
                 <action android:name="android.provider.action.PICK_IMAGES" />
                 <category android:name="android.intent.category.DEFAULT" />
             </intent-filter>
-- 
2.39.0.rc1.256.g54fd8350bd-goog

